### ISSUE TITLE
UnboundLocalError: local variable 'classes' referenced before assignment 

### ISSUE THREAD

**Issue by baditaflorin** [qgis2web/qgis2web#705]
```
An error has occurred while executing Python code: 

UnboundLocalError: local variable 'classes' referenced before assignment 
Traceback (most recent call last):
  File "/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgis2web/maindialog.py", line 335, in saveMap
    feedback=self.feedback)
  File "/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgis2web/leafletWriter.py", line 100, in write
    folder=dest_folder)
  File "/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgis2web/leafletWriter.py", line 268, in writeLeaflet
    useVT, useShapes, useOSMB)
  File "/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgis2web/leafletLayerScripts.py", line 124, in writeVectorLayer
    safeLayerName)
  File "/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgis2web/leafletLayerScripts.py", line 348, in getLegend
    for cnt, c in enumerate(classes):
UnboundLocalError: local variable 'classes' referenced before assignment


Python version: 3.5.2 (default, Nov 23 2017, 16:37:01) [GCC 5.4.0 20160609] 
QGIS version: 3.4.3-Madeira Madeira, 2f64a3c 

Python Path:
/usr/share/qgis/python
/home/florin/.local/share/QGIS/QGIS3/profiles/default/python
/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins
/usr/share/qgis/python/plugins
/usr/lib/python35.zip
/usr/lib/python3.5
/usr/lib/python3.5/plat-x86_64-linux-gnu
/usr/lib/python3.5/lib-dynload
/home/florin/.local/lib/python3.5/site-packages
/usr/local/lib/python3.5/dist-packages
/usr/local/lib/python3.5/dist-packages/setuptools-39.1.0-py3.5.egg
/usr/local/lib/python3.5/dist-packages/pip-10.0.1-py3.5.egg
/usr/local/lib/python3.5/dist-packages/xlsx2csv-0.7.4-py3.5.egg
/usr/lib/python3/dist-packages
/home/florin/.local/share/QGIS/QGIS3/profiles/default/python
/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins/mmqgis/forms
/home/florin/.local/share/QGIS/QGIS3/profiles/default/python/plugins
```

**Comment by tomchadwin** [comment]
How did you trigger this error?

**Comment by tomchadwin** [comment]
Closing without feedback. Comment and I can reopen.

**Comment by HueyNemud** [comment]
Hi,

Got the same error with Leaflet export on QGIS 3.6.1.  
This seems to happen at least in two cases : 
1 - when exporting a vector layer with no symbolizer (but, in my case, with labels)
2 - when using a SVG symbolizer without setting a default SVG symbol (e.g if symbols paths are defined by an attribute).

The OpenLayers export does not trigger python errors but the code produced is incorrect.

How to reproduce:
Create a point vector layer with "no symbol" & export to Leaflet/OpenLayers.

**Comment by kaplanlior** [comment]
@tomchadwin please reopen this issue.

I got this problem when using symbology: point cluster.
I could also reproduce @HueyNemud first point ( didn't try the 2nd one).

**Comment by tomchadwin** [comment]
Reopening after confirming "No symbols" issue. Thanks https://twitter.com/_kyliejane for confirming issue.

**Comment by DanTimmermann** [comment]
I'm having a similar issue. I have been trying to export my map in gqis2web and the error only occurs when I am using point displacement.

**Comment by tomchadwin** [comment]
@DanTimmermann I'm afraid point displacement is unsupported at the moment, yes.

**Comment by DanTimmermann** [comment]
@tomchadwin thanks for responding. I am new to this and wasn't sure if I just really was not getting something. I ended up finding a workaround that worked for my project.

**Comment by UnderJollyRoger** [comment]
@DanTimmermann 
I'm having the same issue with QGIS 3.8. I using point clusters in the map. Could you possibly share your workaround?
![image](https://user-images.githubusercontent.com/32503258/80295705-a0920b00-873a-11ea-9a7a-02803100b920.png)

**Comment by mypitaya** [comment]
An error has occurred while executing Python code: 

UnboundLocalError: local variable 'repo_name' referenced before assignment 
Traceback (most recent call last):
  File "C:\Users/Roel Briones/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qgis_resource_sharing\resource_sharing\gui\resource_sharing_dialog.py", line 281, in edit_repository
    if not repo_name:
UnboundLocalError: local variable 'repo_name' referenced before assignment


Python version: 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)] 
QGIS version: 3.20.0-Odense Odense, decaadbb31 

Python Path:
C:\Users/Roel Briones/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\geo_sim_processing
C:/PROGRA~1/QGIS32~1.0/apps/qgis/./python
C:/Users/Roel Briones/AppData/Roaming/QGIS/QGIS3\profiles\default/python
C:/Users/Roel Briones/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins
C:/PROGRA~1/QGIS32~1.0/apps/qgis/./python/plugins
C:\Program Files\QGIS 3.20.0\apps\Python39\Scripts
C:\Program Files\QGIS 3.20.0\bin\python39.zip
C:\PROGRA~1\QGIS32~1.0\apps\Python39\DLLs
C:\PROGRA~1\QGIS32~1.0\apps\Python39\lib
C:\Program Files\QGIS 3.20.0\bin
C:\PROGRA~1\QGIS32~1.0\apps\Python39
C:\PROGRA~1\QGIS32~1.0\apps\Python39\lib\site-packages
C:\PROGRA~1\QGIS32~1.0\apps\Python39\lib\site-packages\win32
C:\PROGRA~1\QGIS32~1.0\apps\Python39\lib\site-packages\win32\lib
C:\PROGRA~1\QGIS32~1.0\apps\Python39\lib\site-packages\Pythonwin
C:/Users/Roel Briones/AppData/Roaming/QGIS/QGIS3\profiles\default/python
C:\Users\Roel Briones\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\ee_plugin\extlibs_windows
C:\Users\Roel Briones\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\HCMGIS/forms
C:\Users\Roel Briones\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\kmltools\libs
C:\Users/Roel Briones/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qgis2web
C:\Users\Roel Briones\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins
C:\Users\Roel Briones\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\qgis_resource_sharing
C:\Users\Roel Briones\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\qgis_resource_sharing\ext_libs
C:\Users/Roel Briones/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qgis_resource_sharing

Got this problem after the installation of QGIS3.20. 
Somehow affected execution of these apps

Need recommendation badly, thanks a lot.

**Comment by tomchadwin** [comment]
@mypitaya That error message comes from the QGIS Resource Sharing plugin, not from qgis2web. You can raise it on that repo, which might be https://github.com/QGIS-Contribution/QGIS-ResourceSharing (I'm not familiar with that plugin, so I'm not 100% sure).

**Comment by mypitaya** [comment]
Dear Tom,

 I hope you are doing well.
I just followed the instruction from "YOUTUBE Tutorial of jdvkwast as an
instructor of QGIS in IHE-UNESCO,
Using the PCRaster within QGIS, there is a unique processing tool that is
not available in GRASS nor in GDAL
The problem is the tool does not come out in PCRaster Processing Tool,
But it is in the Folders.

Thank you very much for concern.



Kind regards.


Roel
in Davao City

*STAY SAFE*

*ROEL C. BRIONES, C.E.*
8000 Davao City, Mindanao,  Philippines
Telephone:+63082-2332474
Mobile Smart:+63928-4940-497;
            Globe: +63906-444-1027
Email: ***@***.***; ***@***.***

**Comment by Erudition** [comment]
Still getting this error as well.
Whether I'm using an unsupported symbol setup or not, I think this bug is about the UX that there's a Python crash that results, rather that the user being told what's unsupported.

**Comment by andreaordonselli** [comment]
@Erudition Can you upload your project giving the error here (in a zip)?


**This problem is solved:**
`1 - when exporting a vector layer with no symbolr (but, in my case, labels)
`

**I can't reproduce this problem:**
`2 - when using a SVG symbolr without setting a default SVG symbol (e.g. if symbols paths are defined by an attribute).
`

**This problem does not exist because point displacements are not supported**:
`I'm having a similar issue. I have been trying to export my map in gqis2web and the error only occurs when I am using point displacement.`

**This issue affects another plugin**:
`UnboundLocalError: local variable 'repo_name' referenced before assignment`

**Comment by andreaordonselli** [comment]
I have just released version [3.21.0](https://github.com/qgis2web/qgis2web/releases/tag/v3.21.0), which fixes the issues discussed in this thread. The update its just available within the QGIS plugin repository.

Thank you for using the plugin.

I would greatly appreciate it if you would consider donating a virtual coffee through the button on the homepage to support the development of qgis2web.

Best regards.
